pkg/proto/auth/auth.proto

```diff
@@ -27,3 +29,73 @@ message AuthenticationMetadata {
   // be accessed by JMESPathExpressionAuthorizer.
   google.protobuf.Value private = 3;
 }
+
+service Authentication {
+  // Authenticates a request that the client has received.
+  rpc Authenticate(AuthenticateRequest) returns (AuthenticateResponse) {}
+}
+
+message AuthenticateRequest {
+  message ValueList {
+    repeated string value = 1;
+  }
+
+  // The metadata is forwarded as part of the request as accessing the
+  // authentication service might have its own authentication and authorizing
+  // system.
+  map<string, ValueList> request_metadata = 1;
```

Comment:
So this is the equivalent of HTTP or gRPC request headers? Maybe good to explicitly state that?